db/template_test.go

```diff
@@ -364,6 +433,173 @@ func TestGetTemplateWithInvalidID(t *testing.T) {
+		revisions = revisions + 1
+		return nil
+	})
+	if err != nil {
+		t.Error(err)
+	}
+	if revisions != 1 {
+		t.Errorf("expected 1 template revisions stored in the database but we got %d", revisions)
+	}
+
+	// update template
+	w.Name = "updated-template-name"
+	data, err := json.Marshal(w)
+	if err != nil {
+		t.Error(err)
+	}
+	err = tinkDB.UpdateTemplate(ctx, w.Name, string(data), uuid.MustParse(w.ID))
+	if err != nil {
+		t.Error(err)
+	}
+
+	// revalidate revisions
+	revisions = 0
+	err = tinkDB.ListRevisionsByTemplateID(w.ID, func(revision int, tCr *timestamp.Timestamp) error {
+		revisions = revisions + 1
+		return nil
+	})
+	if err != nil {
+		t.Error(err)
+	}
+	if revisions != 2 {
+		t.Errorf("expected 2 template revisions stored in the database but we got %d", revisions)
+	}
+}
+
+func TestTemplateRevisionMigration(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	db, tinkDB, cl := NewPostgresDatabaseClient(t, ctx, NewPostgresDatabaseRequest{})
+	defer func() {
+		err := cl()
+		if err != nil {
+			t.Error(err)
+		}
+	}()
+
+	// apply old migrations
+	_, err := migrate.Exec(db, "postgres", migrate.MemoryMigrationSource{
+		Migrations: []*migrate.Migration{
+			migration.Get202009171251(),
+			migration.Get202010071530(),
+			migration.Get202010221010(),
+			migration.Get202012041103(),
+			migration.Get202012091055(),
+			migration.Get2020121691335(),
+		},
+	}, migrate.Up)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// create a few templates
+	err = createTemplatesFromSQL(ctx, db)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// check if the templates have been created successfully
+	count := 0
+	err = tinkDB.ListTemplates("%", func(id, n string, in, del *timestamp.Timestamp) error {
+		count = count + 1
+		return nil
+	})
+	if err != nil {
+		t.Error(err)
+	}
+	if count != 5 {
+		t.Errorf("expected %d templates stored in the database but we got %d", 5, count)
+	}
+
+	// apply template-revision migration
+	_, err = migrate.Exec(db, "postgres", migrate.MemoryMigrationSource{
+		Migrations: []*migrate.Migration{
+			migration.Get202009171251(),
+			migration.Get202010071530(),
+			migration.Get202010221010(),
+			migration.Get202012041103(),
+			migration.Get202012091055(),
+			migration.Get2020121691335(),
+			migration.Get202102111035(),
+		},
+	}, migrate.Up)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// validate if the migration was successful
+	revisionCount := 0
+	err = tinkDB.ListTemplates("%", func(id, n string, in, del *timestamp.Timestamp) error {
+		err := tinkDB.ListRevisionsByTemplateID(id, func(revision int, tCr *timestamp.Timestamp) error {
+			revisionCount = revisionCount + 1
+			return nil
+		})
+		return err
+	})
+	if err != nil {
+		t.Error(err)
+	}
+	if revisionCount != 5 {
+		t.Errorf("expected %d template revisions stored in the database but we got %d", 5, revisionCount)
+	}
+}
+
+func createTemplatesFromSQL(ctx context.Context, db *sql.DB) error {
```

Comment:
Can you add a comment? I am sure the next person or even myself in a couple of days will not remember why we have this function and the name is so generic that it won't help